lib/pos_mempool.go

```diff
@@ -999,6 +999,28 @@ func (mp *PosMempool) pruneNoLock() error {
 	return nil
 }
 
+func (mp *PosMempool) rebucketTransactionRegister() error {
+	// Check if the global params haven't changed in a way that requires rebucketing the
+	// transaction register
+	if mp.txnRegister.HasGlobalParamChange(mp.globalParams) {
+		return nil
+	}
+
+	// Rebucket the transaction register
+	newTxnRegister, err := mp.txnRegister.CopyWithNewGlobalParams(mp.globalParams)
```

Comment:
do we need to grab a read lock on the `txnRegister` here? Okay just checked, `CopyWithGlobalParams` acquires the lock for us.